resources/idlharness.js

```diff
@@ -1828,28 +1828,28 @@ IdlInterface.prototype.test_self = function()
                 // only way to get our hands on it. We only test that its class
                 // string, at least, is correct.
                 assert_class_string(Object.getPrototypeOf(this.get_interface_object().prototype),
-                                    inherit_interface + 'Prototype',
+                                    inherit_interface,
                                     'Class name for prototype of ' + this.name +
-                                    '.prototype is not "' + inherit_interface + 'Prototype"');
+                                    '.prototype is not "' + inherit_interface + '"');
             }
         }
 
         // "The class string of an interface prototype object is the
-        // concatenation of the interface’s qualified identifier and the string
-        // “Prototype”."
+        // interface’s qualified name."
 
-        // Skip these tests for now due to a specification issue about
-        // prototype name.
-        // https://www.w3.org/Bugs/Public/show_bug.cgi?id=28244
+        // "If an object has a class string classString, then the object must,
+        // at the time it is created, have a property whose name is the
+        // @@toStringTag symbol with PropertyDescriptor{[[Writable]]: false,
+        // [[Enumerable]]: false, [[Configurable]]: true, [[Value]]: classString}."
 
-        // assert_class_string(this.get_interface_object().prototype, this.get_qualified_name() + "Prototype",
-        //                     "class string of " + this.name + ".prototype");
+        assert_class_string(this.get_interface_object().prototype, this.get_qualified_name(),
+                            "class string of " + this.name + ".prototype");
 
         // String() should end up calling {}.toString if nothing defines a
         // stringifier.
         if (!this.has_stringifier()) {
```

Comment:
If the above check of a `Symbol.toStringTag` own property is done, then I think this whole bit can just be removed. The presence of a `Symbol.toStringTag` doesn't depend on whether there's a stringifier, and can be tested even in the presence of a stringifier. We don't need to also test that `String()` is implemented correctly, at least not for the prototype object.